### DISCUSSION TITLE
Is it possible to get true color in the TTY?

### DISCUSSION THREAD

**Discussion by char-png** [browsh-org/browsh#481] [Q&A] [answered]
I've been trying to use Browsh on my old laptop running a very minimal Arch with the zsh shell if that matters and it works perfectly fine* but I can't get colors to work properly :(

*WHEN firefox doesn't take a whole minute to open, but that's my laptops fault :p, is there any way to make the 30 second timeout longer?

I've looked everywhere and nothing I've tried has worked so far but I've been jumping from link to link for hours so maybe I just missed something obvious

By the way I also have another issue, since I'm in the CLI I need to use gpm to be able to select and click around but apparently clicking with gpm doesn't work on Browsh... from what I've seen that's the fault of one of it's dependencies and Browsh might have a vim-mode in the future. So if anyone knows a workaround or if the vim-mode will come somewhat "soon" I'd like to know!

**Comment by tombh** [comment]
I wonder if you could start a headless Firefox yourself and then tell Browsh to connect to it with the `--firefox.use-existing` flag?

Yeah, the Vim support has been in the works for years 🫤 https://github.com/browsh-org/browsh/pull/439

**Reply by char-png** [reply]
I can and when I do and then try to launch Browsh without the tag it says there's a headless firefox running already but if I use the tag it says it couldn't connect to it in 30 seconds.

So I just need to keep starting and killing firefox and trying both ways of opening Browsh until it works, it's always after killing firefox and then starting Browsh with the use-existing tag somehow... But then it stays at "Waiting for a user-initiated Firefox instance to connect..." so I close it and launch Browsh normally and THEN it opens

> Yeah, the Vim support has been in the works for years 🫤 https://github.com/browsh-org/browsh/pull/439

Aw so it's still far from done? Honestly if it's somewhat functional I'd be fine with it even if it's buggy or missing features because right now I've got no way at all of clicking buttons and links :p Is there someway to patch my Browsh to use the WIP vim mode? or a patch for gpm to work on Browsh?

Also since you didn't answer my initial question I suppose you don't know of a way to get true color in the tty?

**Comment by tombh** [comment, marked as answer]
Oh, that's really frustrating about the manual headless Firefox, sorry. Maybe it's the timing? Like you have to wait a minute or Firefox to boot up before starting Browsh?

The Vim branch is actually working, it's the broken tests that prevent it from being merged. So you could checkout the branch and build your own version of Browsh. In fact I did that when I made that PR, and it was working well.

Oh, right. I'm afraid I don't know about true colour TTYs. My guess is that true colour is the domain of Xorg/Wayland etc and so TTYs aren't going to be so motivated to support it.

**Reply by char-png** [reply]
> Maybe it's the timing?

Nope, I tried waiting in all of the steps and even if it boots it just wants to see me suffer :p

> The Vim branch is actually working

Great! I might try it but actually I think I'll just use a simple window manager and see if that solves my issue with colors and the clicking. But first I'll fully reinstall Arch because I could have sworn it wasn't THIS slow last time I used it... I likely did something wrong not gonna lie :p

> My guess is that true colour is the domain of Xorg/Wayland etc

Yeah I've lost all hope already :( But I tried Browsh on my desktop and it looks AWESOME!! I've had some issues like not being able to play youtube videos and the search bar in peertube not working at all but it works really well for the websites I normally use!

Oh by the way, is there a reason why you made a new comment instead of a reply to the first comment? Am I supposed to use comments instead of replies? I've never used github discussions before and didn't even know they existed LOL!

**Comment by tombh** [comment]
BTW I use Arch too 😏

I'm sorry Browsh still has so many rough edges. Though I'm pleased it can still make people go AWESOME haha.

Did you see the other terminal browser, it's does some things a lot better than Browsh: https://github.com/fathyb/carbonyl

**Reply by char-png** [reply]
Yeah I've tried it too! Browsh seems to load pages faster and it's already slow enough :P But I'm keeping both just in case there are any websites that work on one but not the other!

Btw I reinstalled Arch and everything works fine now :)
